src/script/interpreter.cpp

```diff
@@ -1173,8 +1195,18 @@ PrecomputedTransactionData::PrecomputedTransactionData(const CTransaction& txTo)
     hashOutputs = GetOutputsHash(txTo);
 }
 
-uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, SigVersion sigversion, const PrecomputedTransactionData* cache)
+uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, SigVersion sigversion, const PrecomputedTransactionData* cache,int selected_forkid)
 {
+    int nForkSigHashType=nHashType; // see item 10 in https://github.com/Bitcoin-UAHF/spec/blob/master/replay-protected-sighash.md
+    /* Two-way replay protection in BCH was implemented with a quadratic scaling fix replicating a lot of segwit's functionality.  They change the encoded versionbits inside the transaction hash (in addition to their
+       own 'segwit-lite' implementation.  Much/most of the replay-protected modified sighash in BCH is actually implementing that functionality.
+       However, because BTCG has segwit already, that fix is unnecessary.  The only thing necessary is to add the forkid to the nHashType in the hash to change the transaction hash.
+    */
+    if(nHashType & SIGHASH_FORKID)
```

Comment:
It would be nice if forkid replay protected transactions always use SIGVERSION_WITNESS_V0 signatures even for non-segwit transactions, just like in BCH. This makes it easier for hardware wallet especially if they already support BCH.

Is there a reason not to use SIGVERSION_WITNESS_V0 signatures for everything (and get rid of the code for old-style signatures)?  Even Johnson Lau's forcenet suggested this for replay protection: https://github.com/jl2012/bitcoin/commit/7c2593946c4f3e210683110782d82f55473c682a#diff-be2905e2f5218ecdbe4e55637dac75f3L1188

Reply:
You are right. We are currently working on that but it's targeted for the next testnet launch (the testnet v1 will be reset to change the SIGHASH method)

Reply:
@h4x3rotab To confirm: the mainnet will use `SIGVERSION_WITNESS_V0` for non-SegWit transactions?

Reply:
Yes. Track the progress here: https://github.com/BTCGPU/BTCGPU/pull/150